imports/server/gdriveActivityFetcher.ts

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable no-await-in-loop */
+import { Meteor } from 'meteor/meteor';
+import Flags from '../Flags';
+import { ACTIVITY_GRANULARITY, ACTIVITY_SEGMENTS } from '../lib/config/activityTracking';
+import DocumentActivities from '../lib/models/DocumentActivities';
+import Documents from '../lib/models/Documents';
+import MeteorUsers from '../lib/models/MeteorUsers';
+import roundedTime from '../lib/roundedTime';
+import GoogleClient from './googleClientRefresher';
+import ignoringDuplicateKeyErrors from './ignoringDuplicateKeyErrors';
+import DriveActivityLatests from './models/DriveActivityLatests';
+import Locks, { PREEMPT_TIMEOUT } from './models/Locks';
+
+async function recordDriveChanges(ts: Date, fileIds: string[], googleAccountIds: string[]) {
+  const time = roundedTime(ACTIVITY_GRANULARITY, ts);
+
+  // In all likelihood, we will only have one of each of these, but for
+  // completeness we'll record the full cartesian product
+  for await (const fileId of fileIds) {
+    const document = await Documents.findOneAsync({ 'value.id': fileId });
+    if (!document) {
+      // eslint-disable-next-line no-continue
+      continue;
+    }
+
+    for await (const googleAccountId of googleAccountIds) {
+      // if we can't look up the account ID, we'll record as user=undefined
+      // which we'll count as a seprate user
+      const user = await MeteorUsers.findOneAsync({ googleAccountId, hunts: document.hunt });
```

Comment:
We can sort by `createdAt` so that we'll pick a consistent user, so long as they don't link/unlink their account (in which case I think anything goes, but that seems acceptable)